functions/src/controllers/AdminController.ts

```diff
@@ -208,26 +225,339 @@ const Admin = {
+            if (!businessData.airtable_config.base_id) {
+                res.status(500).send({ message: "No config found for airtable, Please contact admin for more information" });
+            }
+
+            const base_id = businessData.airtable_config.base_id;
+            const base = Airtable.base(base_id);
+            const dataMaster = {}
+            const productMaster = {}
+            base("external_orders").select({
+                view: "Orders To Upload",
+                pageSize: 5
+            }).eachPage(function page(records, fetchNextPage) {
+                console.log(records.length);
+
+                records.forEach(async function (record) {
+                    let recordFieldKeys = Object.keys(record.fields)
+                    const checkDifference = _.difference(EXTERNAL_ORDERS_MANDATORY_FIELDS,recordFieldKeys)
+
+                    if (!checkDifference.length) {
+                        const orderObj = record.fields
+                        orderObj.airtableId = record.id
+                        if (dataMaster[orderObj.order_id]) {
+                            dataMaster[orderObj.order_id].push(orderObj)
+                        } else {
+                            dataMaster[orderObj.order_id] = []
+                            dataMaster[orderObj.order_id].push(orderObj)
+                        }
+                        if (productMaster[orderObj.product_id]) {
+                            productMaster[orderObj.product_id].push({ variant_id: orderObj.variant_id })
+                        } else {
+                            productMaster[orderObj.product_id] = []
+                            productMaster[orderObj.product_id].push({ variant_id: orderObj.variant_id })
+                        }
+                    } else {
+                        console.log("validation fale");
+
+                    }
+
+                });
+
+                fetchNextPage();
+
+            }, async function done(err) {
+                if (err) { console.error(err); return res.status(500).send({ err }) }
+                for (const productId in productMaster) {
+                    console.log(productId);
+
+                    try {
+                        const prodRef = await db.collection('products').doc(productId).get()
+                        if (prodRef.exists) {
+                            productMaster[productId] = prodRef.data();
+                        }
+                    } catch (e) {
+                        console.log(e)
+                    }
+                }
+                //build order object
+
+                const ordermaster = []
+                console.log(Object.keys(dataMaster).length);
+                const userStore = {}
+                for (let ex_order_id in dataMaster) {
+                    console.log("processing order",ex_order_id);
+                    
+                    let name,email = ""
+                    let orderObj = {
```

Comment:
building of order object should be moved to util method for orders - createExternalOrderObject(). This is too much of inline code